object/virtual_machine.go

```diff
@@ -648,6 +648,24 @@ func (v VirtualMachine) CreateSnapshot(ctx context.Context, name string, descrip
 	return NewTask(v.c, res.Returnval), nil
 }
 
+// CreateSnapshotEx creates a new snapshot of a virtual machine.
+func (v VirtualMachine) CreateSnapshotEx(ctx context.Context, name string, description string, memory bool, quiesceSpec *types.VirtualMachineGuestQuiesceSpec) (*Task, error) {
```

Comment:
`quiesceSpec` should be `types.BaseVirtualMachineGuestQuiesceSpec` so [VirtualMachineWindowsQuiesceSpec](https://developer.broadcom.com/xapis/vsphere-web-services-api/latest/vim.vm.WindowsQuiesceSpec.html) can also be passed here.

Tempting to change the existing `CreateSnapshot` to call `CreateSnapshotEx`, and make the breaking change to the `quiesceSpec` param (bool -> types.BaseVirtualMachineGuestQuiesceSpec). I think govmomi's use of `CreateSnapshot` goes back to vSphere 5.0, `CreateSnapshotEx` was added in 6.5 and we're not supporting vSphere < 7.0 in govmomi anymore.

Reply:
Thanks Doug. I've made the change to types.BaseVirtualMachineGuestQuiesceSpec. Good catch on the WindowsQuiesceSpec.

As for changing the existing CreateSnapshot to call CreateSnapshotEx. From looking at our VMODL docs, CreateSnapshot wasn't deprecated until 8.0 GA.

Reply:
ok thanks, let's leave existing CreateSnapshot as-is then